### PR TITLE
feat(helpers): Stop reporting errors after a param add/swap is fixed

### LINKED ISSUE #421: Stop reporting lint errors for param additions/swaps that still have already been fixed
Currently the param addition and swap codemods still report linting errors even after the event has already been added/moved.

Ideally I think this wouldn't happen, as it causes some clutter in the linter output that I don't think is actually helpful for consumers. It definitely wasn't helpful when I was doing the topology v5 update as it just made it harder to actually know if everything had been addressed.

### PR DESCRIPTION
**Pull request by wise-king-sullyman**
Closes #421

### CODE REVIEW

**Comment by wise-king-sullyman** [comment]
Helpful command for testing this in test.tsx:
`yarn test:v5:single --only applicationLauncher-updated-params,calendarMonth-update-onChange-onMonthChange-params,clipboardCopy-onChange-event-added,dataList-updated-callback,datalist-remove-selectableRow,dataListCheck-updated-callback,drawerPanelContent-updated-callback`

File: packages/eslint-plugin-pf-codemods/lib/helpers.js [hunk 1]
```diff
@@ -493,15 +493,26 @@ function addCallbackParam(componentsArray, propMap) {
...
                 const { type, params } = propProperties;
 
+                const parameterConfig = propMap[attribute.name.name];
+
+                const isParamAdditionOnly = typeof parameterConfig === "string";
+                const newOrDefaultParamName = isParamAdditionOnly ? parameterConfig : parameterConfig.defaultParamName;
+                const firstParamName = params && params[0]?.name;
+
+                // if the first parameter is already the expected "fixed" result, early return with no error
+                if (firstParamName === newOrDefaultParamName || !isParamAdditionOnly && firstParamName?.match(parameterConfig.otherMatchers)) {
```
[the remark this answers is not in the document]
Comment (wise-king-sullyman) [reply packages/eslint-plugin-pf-codemods/lib/helpers.js:503]: Yeah I could see the utility there

**Review by gitdallas — APPROVED** [review]
lgtm, trying to guess what a user might do is difficult, but eric's comment would be easy enough to implement and could be helpful

**Comment by wise-king-sullyman** [comment]
It wasn't as easy as you probably would've expected 😆 definitely wasn't as easy as I expected. @thatblindgeye's suggestion is done though.

File: packages/eslint-plugin-pf-codemods/lib/helpers.js [hunk 2]
```diff
@@ -493,15 +501,33 @@ function addCallbackParam(componentsArray, propMap) {
                 }
                 const { type, params } = propProperties;
 
+                const parameterConfig = propMap[attribute.name.name];
+                const isParamAdditionOnly = typeof parameterConfig === "string";
+                const newOrDefaultParamName = isParamAdditionOnly ? parameterConfig : parameterConfig.defaultParamName
+
+                let potentialParamMatchers = `(_?${stripLeadingUnderscore(newOrDefaultParamName)})`
+                const otherMatchersString = parameterConfig.otherMatchers?.toString().slice(1, -1)
```
Comment (gitdallas) [inline packages/eslint-plugin-pf-codemods/lib/helpers.js:509]:
```suggestion
                const otherMatchersString = parameterConfig.otherMatchers?.toString().slice(1, -1);
```

Comment (wise-king-sullyman) [reply packages/eslint-plugin-pf-codemods/lib/helpers.js]: Prettier did a lot of things that I don't love formatting wise, but your semicolons are in now 🤣

**Review by gitdallas — APPROVED** [review]

**Pull request merged**